src/backend/distributed/commands/table.c

```diff
@@ -194,6 +200,14 @@ PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
 static void
 PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 {
+	if (!ShouldEnableLocalReferenceForeignKeys())
+	{
+		/*
+		 * Either the user disabled foreign keys from/to local/reference tables
+		 * or the coordinator is not in the metadata */
+		return;
+	}
+
```

Comment:
we shouldn't need this as `alter table` hook would already check

Reply:
I keep it just to make the flow easier to follow and robust for future changes.

Does that sound good?